src/types/timestamp.rs

```diff
@@ -1136,7 +1098,7 @@ impl From<DateTime<FixedOffset>> for Timestamp {
         // Get a copy of the offset to store separately
         let offset = Some(*fixed_offset_date_time.offset());
         let precision = Precision::Second;
-        let fractional_seconds = Some(Mantissa::Digits(9));
+        let fractional_seconds = Mantissa::Digits(0);
```

Comment:
Types from the `chrono` crate (`NaiveDateTime`, `DateTime<FixedOffset>`) always use nanosecond precision, which is why this (and the one above on what is now line 1088) were using `Digits(9)`. Was this an intentional change?

Reply:
Yes- basically, the `test_timestamp_fractional_seconds_scale` unit test would break since the builder would use a `DateTime<FixedOffset>` to create the `Timestamp`, and `From<DateTime<FixedOffset>>` would default `fractional_seconds` to nanosecond precision.

Should we revert this to nanosecond precision and construct `timestamp_with_seconds` with `.with_nanoseconds_and_precision(0, 0)` instead?

Reply:
I think this should remain `Digits(9)` since someone converting from `DateTime<FixedOffset>` to `Timestamp` in other contexts would expect the nanoseconds to be preserved.

In the unit test you mention: 

https://github.com/amzn/ion-rust/blob/fb164687c5dca001b144c4fe36f8edd06c5d4556/src/types/timestamp.rs#L1568-L1569

it should be that `with_ymd_hms` sets the `Timestamp`'s precision to `Second` and its mantissa to `Digits(0)` after constructing the `DateTime<FixedOffset>`.